### PR TITLE
Removed unneeded deserialization and made key&value deserialization l…

### PR DESCRIPTION
**Pull request by ahmetmircik**
…azy in ReplicatedMapEntryView

fixes https://github.com/hazelcast/hazelcast/issues/14235

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/ReplicatedMapEntryView.java [hunk 1]
```diff
@@ -33,21 +34,31 @@
...
     }
 
+    public ReplicatedMapEntryView(SerializationService serializationService) {
+        this.serializationService = serializationService;
+    }
+
     @Override
     public K getKey() {
-        return key;
+        // `null` `serializationService` means, use raw type without any conversion
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/ReplicatedMapEntryView.java]: minor: markdown markup in comment is not respected unfortunately... (I also do this all the time :-) )

File: hazelcast-client/src/test/java/com/hazelcast/client/replicatedmap/ClientReplicatedMapTest.java [hunk 2]
```diff
@@ -562,6 +566,104 @@ public void remove_empties_internal_ttl_schedulers() {
...
+        replicatedMap.remove(key);
+
+        assertEquals(0, ((Key1) key).COUNTER.get());
+        // expect only 1 deserialization in ClientReplicatedMapProxy#remove method
+        assertEquals(1, ((Value1) value).COUNTER.get());
+    }
+
+    @Test
+    public void no_key_value_deserialization_on_server_when_entry_is_get() {
+        if (inMemoryFormat == InMemoryFormat.OBJECT) {
```
[the remark this answers is not in the document]
Comment (ahmetmircik) [reply hazelcast-client/src/test/java/com/hazelcast/client/replicatedmap/ClientReplicatedMapTest.java]: done

File: hazelcast-client/src/test/java/com/hazelcast/client/replicatedmap/ClientReplicatedMapTest.java [hunk 3]
```diff
@@ -562,6 +566,104 @@ public void remove_empties_internal_ttl_schedulers() {
         assertAllTtlSchedulersEmpty(node.getReplicatedMap(mapName));
     }
 
+    @Test
+    public void no_key_value_deserialization_on_server_when_entry_is_removed() {
+        if (inMemoryFormat == InMemoryFormat.OBJECT) {
```
[the remark this answers is not in the document]
Comment (ahmetmircik) [reply hazelcast-client/src/test/java/com/hazelcast/client/replicatedmap/ClientReplicatedMapTest.java]: done

File: hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/ReplicatedMapEntryView.java (hunk 1 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/ReplicatedMapEntryView.java]: copy&paste error. fixed.

File: hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/ReplicatedMapEntryView.java [hunk 4]
```diff
@@ -183,8 +198,8 @@ public int getId() {
     @Override
     public String toString() {
         return "ReplicatedMapEntryView{"
-                + "key=" + key
-                + ", value=" + value
+                + "key=" + getKey()
+                + ", value=" + getValue()
                 + ", creationTime=" + creationTime
```
[the remark this answers is not in the document]
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/ReplicatedMapEntryView.java]: seems out of scope now, we can add it later.

**Pull request merged**